### PR TITLE
Suggestion for catching shape error in Jacobian computations

### PR DESCRIPTION
**Pull request by joelnmdyer**
Suggestion for closing issue #51

### CODE REVIEW

**Comment by joelnmdyer** [comment]
Perhaps even better to just have something that checks for Jacobians that are >1D and says `Please ensure your loss function produces a scalar (i.e. 0D) torch.tensor` or something

**Comment by arnauqb** [comment]
Is there a use case where we would expect the loss value to not be a scalar? If there isn't then it is probably easier to just enforce the user to return a scalar loss.

**Comment by joelnmdyer** [comment]
I don't think you'd want a loss that wasn't scalar, no... This commit I think should fix it, checking for scalar loss before running VI

**Comment by arnauqb** [comment]
looks good to me! happy for you to merge it
